ChaosMod/Effects/db/Player/PlayerAimbot.cpp

```diff
@@ -0,0 +1,37 @@
+/*
+	Effect by Last0xygen
+*/
+
+#include <stdafx.h>
+
+static void OnStop()
+{
+	SET_PED_INFINITE_AMMO_CLIP(PLAYER_PED_ID(), false);
+}
+
+static void OnTick()
+{
+	Ped playerPed = PLAYER_PED_ID();
+	SET_PED_INFINITE_AMMO_CLIP(playerPed, true);
+	if (IS_PED_WEAPON_READY_TO_SHOOT(playerPed))
+	{
+		float maxRange = GET_MAX_RANGE_OF_CURRENT_PED_WEAPON(playerPed);
+		Vector3 playerCoord = GET_ENTITY_COORDS(playerPed, false);
+		for (Ped ped : GetAllPeds())
+		{
+			REQUEST_PED_VISIBILITY_TRACKING(ped);
+			if (ped != playerPed && IS_TRACKED_PED_VISIBLE(ped) && !IS_PED_DEAD_OR_DYING(ped, true) && IS_PED_HUMAN(ped) && !IS_PED_A_PLAYER(ped) && IS_TRACKED_PED_VISIBLE(ped))
+			{
+				Vector3 pedCoord = GET_ENTITY_COORDS(ped, 0);
+				float distance = GET_DISTANCE_BETWEEN_COORDS(playerCoord.x, playerCoord.y, playerCoord.z, pedCoord.x, pedCoord.y, pedCoord.z, true);
+				if (distance <= maxRange)
+				{
+					Vector3 headVector = GET_PED_BONE_COORDS(ped, 0x796E, 0, 0, 0); // Head
+					SET_PED_SHOOTS_AT_COORD(playerPed, headVector.x, headVector.y, headVector.z, true);
```

Comment:
Could add `TASK_AIM_GUN_AT_ENTITY(playerPed, ped, 1, false);` here to make the character aim at the targets

Reply:
although it makes the walking animation a bit janky

Reply:
yeah, the walking is hardly working if the aim-task has been set (i.e. walks forwards even if only sidesteps pressed)
-> Don't want to add that 😅